src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java

```diff
@@ -408,12 +409,12 @@ public void queueMove(Location from, Location to) {
         Validate.notNull(to, "Target Location cannot be null!");
 
         // This collection is iterated over in a different thread. Need to lock it.
-        queueLock.readLock().lock();
+        queueLock.writeLock().lock();
 
         try {
             movingQueue.put(from, to);
         } finally {
-            queueLock.readLock().unlock();
+            queueLock.writeLock().unlock();
```

Comment:
I feel like we should make a LockableCollection so that we can make sure things are locked properly at all times.

`map.put(key, value)`
will actually run like

```java
lock.writeLock().lock();
try {
    realCollection.put(key, value);
} finally {
    lock.writeLock().unlock();
}
```

Reply:
Isn't that what `Collections#synchronizedX` does?

Reply:
Yeah Collections does have synchronized things. I do prefer synchronized here since we're locking in a very simple way but I didn't want to change everything right away 😆

Reply:
ah fair enough